src/org/zaproxy/zap/extension/ascan/ActiveScan.java

```diff
@@ -262,12 +263,18 @@ public void notifyNewMessage(final HttpMessage msg) {
 		}
 		
 		this.rcTotals.incResponseCodeCount(msg.getResponseHeader().getStatusCode());
-		
-        if (hRef != null && this.rcTotals.getTotal() <= this.maxResultsToList) {
-            // Very large lists significantly impact the UI responsiveness
-            // limiting them makes large scans _much_ quicker
-        	addHistoryReference(hRef);
-    	}
+
+		if (hRef != null) {
+			// Very large lists significantly impact the UI responsiveness
+			// limiting them makes large scans _much_ quicker
+			if (this.rcTotals.getTotal() <= this.maxResultsToList) {
+				addHistoryReference(hRef);

```

Comment:
This statement could be moved to outside the then/else blocks (it's done always). We just need to invert the expression (and delete in the then block).